### PR TITLE
DRILL-4764: Parquet file with INT_16, etc. logical types not supporte…

### PR DESCRIPTION
**Pull request by Serhii-Harnyk**
…d by simple SELECT

### CODE REVIEW

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet2/TestDrillParquetReader.java [hunk 1]
```diff
@@ -84,4 +94,43 @@ public void test4349() throws Exception {
...
+      .withType(schema)
+      .withConf(conf)
+      .build();
+
+    SimpleGroupFactory groupFactory = new SimpleGroupFactory(schema);
+    writer.write(
+      groupFactory.newGroup()
+        .append("int8_field", -128)
+        .append("uint8_field", 127)
+        .append("int16_field", -32768)
```
Comment (parthchandra) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet2/TestDrillParquetReader.java]: Can you add test values for unsigned types as well? 0, -1, 256, 65536, etc

**Comment by Serhii-Harnyk** [comment]
@parthchandra, I have made some changes and added few test values. Could you please review it?

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java [hunk 2]
```diff
@@ -56,6 +58,31 @@ protected void readField(long recordsToReadInThisPass) {
...
+    // this method is called by its superclass during a read loop
+    @Override
+    protected void readField(long recordsToReadInThisPass) {
+
+      recordsReadInThisIteration = Math.min(pageReader.currentPageCount
+        - pageReader.valuesRead, recordsToReadInThisPass - valuesReadInCurrentPass);
+
+      if (usingDictionary) {
+        for (int i = 0; i < recordsReadInThisIteration; i++){
+          valueVec.getMutator().setSafe(valuesReadInCurrentPass + i, pageReader.dictionaryValueReader.readInteger());
```
Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: you probably need to set writer index just as you do in the uint8 case.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java [hunk 3]
```diff
@@ -56,6 +58,31 @@ protected void readField(long recordsToReadInThisPass) {
     }
```
Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: What about NullableDictionaryUint types?
Also, It is probably a good idea to create a single parquet file with columns of all combinations of these types, 
Nullable, dictionary encoded, signed and unsigned int2, int4, int8 types and a single unit test to read that file.

**Comment by Serhii-Harnyk** [comment]
@parthchandra, could you please review new changes?

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java (hunk 2 above)
Comment (Serhii-Harnyk) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: done

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java (hunk 3 above)
Comment (Serhii-Harnyk) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: done

**Comment by parthchandra** [comment]
+1

**Comment by Serhii-Harnyk** [comment]
Squashed all changes into single commit and rebased on master

**Pull request closed without merging**